packages/web3-providers-ws/src/index.ts

```diff
@@ -0,0 +1,387 @@
+				this.sentQueue.delete(key);
+			});
+		}
+
+		if (
+			!this.wsProviderOptions?.reconnectOptions?.maxAttempts ||
+			this.reconnectAttempts < this.wsProviderOptions.reconnectOptions.maxAttempts
+		) {
+			setTimeout(() => {
+				this.reconnectAttempts += 1;
+
+				this.removeSocketListeners();
+				this.connect();
+			}, this.wsProviderOptions?.reconnectOptions?.delay);
+		}
+	}
+
+	private onMessage(e: IMessageEvent): void {
+		this.parseResponse(typeof e.data === 'string' ? e.data : '').forEach(
+			(response: JsonRpcResponse | SubscriptionResultNotification) => {
+				if ('method' in response && response.method.endsWith('_subscription')) {
+					this.wsEventEmitter.emit('message', null, response);
+					return;
+				}
+
+				const { id } = response;
+
+				if (id && this.sentQueue.has(id)) {
+					const requestItem = this.sentQueue.get(id);
+					if ('result' in response && response.result !== undefined) {
+						this.wsEventEmitter.emit('message', null, response);
+						requestItem?.deferredPromise.resolve(response);
+					} else if ('error' in response && response.error !== undefined) {
+						this.wsEventEmitter.emit('message', response, null);
+						requestItem?.deferredPromise.reject(response);
+					}
+
+					this.sentQueue.delete(id);
+				}
+			},
+		);
+	}
+
+	private onConnect() {
+		this.providerStatus = 'connected';
+		this.reconnectAttempts = 0;
+
+		if (this.requestQueue.size > 0) {
+			for (const value of this.requestQueue.values()) {
+				// eslint-disable-next-line
+				this.request(value.payload);
+			}
+		}
+	}
+
+	private onClose(event: ConnectionEvent): void {
+		this.providerStatus = 'disconnected';
+		if (
+			this.wsProviderOptions?.reconnectOptions?.auto &&
+			(![1000, 1001].includes(event.code) || event.wasClean === false)
+		) {
+			this.reconnect();
+			return;
+		}
+
+		if (this.requestQueue.size > 0) {
+			this.requestQueue.forEach((request: WSRequestItem, key: JsonRpcId) => {
+				request.deferredPromise.reject(new ConnectionNotOpenError(event));
+				this.requestQueue.delete(key);
+			});
+		}
+
+		if (this.sentQueue.size > 0) {
+			this.sentQueue.forEach((request: WSRequestItem, key: JsonRpcId) => {
+				request.deferredPromise.reject(new ConnectionNotOpenError(event));
+				this.sentQueue.delete(key);
+			});
+		}
+
+		this.removeSocketListeners();
+	}
+
+	private parseResponse(dataReceived: string): JsonRpcResponse[] {
+		const returnValues: JsonRpcResponse[] = [];
+
+		// de chunk
+		const dechunkedData = dataReceived
+			.replace(/\}[\n\r]?\{/g, '}|--|{') // }{
+			.replace(/\}\][\n\r]?\[\{/g, '}]|--|[{') // }][{
+			.replace(/\}[\n\r]?\[\{/g, '}|--|[{') // }[{
+			.replace(/\}\][\n\r]?\{/g, '}]|--|{') // }]{
+			.split('|--|');
+
+		dechunkedData.forEach((data: string) => {
+			let dataToParse: string = data;
+
+			if (this.lastDataChunk !== '') dataToParse = this.lastDataChunk + dataToParse;
+
+			let result: JsonRpcResponse;
+
+			try {
+				// eslint-disable-next-line
```

Comment:
I think an explanation (like error message + reasons why it's not being fixed) should be provided when disabling linting. Additionally, you should be able to disable the specific rule for the line instead of disabling linting entirely